gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/sideEffect/Aggregate.feature

```diff
@@ -144,145 +144,490 @@ Feature: Step - aggregate()
+      | result |
+      | d[27].i |
+
+  Scenario: g_withSideEffectXa_1_maxX_V_aggregateXaX_byXageX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", 1, Operator.max).V().aggregate("a").by("age").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+    | result |
+    | d[35].i |
+
+  Scenario: g_withSideEffectXa_1_maxX_V_aggregateXlocal_aX_byXageX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", 1, Operator.max).V().aggregate(Scope.local, "a").by("age").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | d[35].i |
+
+  Scenario: g_withSideEffectXa_100_maxX_V_aggregateXaX_byXageX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", 100, Operator.max).V().aggregate("a").by("age").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+    | result |
+    | d[100].i |
+
+  Scenario: g_withSideEffectXa_100_maxX_V_aggregateXlocal_aX_byXageX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", 100, Operator.max).V().aggregate(Scope.local, "a").by("age").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | d[100].i |
+
+  Scenario: g_withSideEffectXa_true_andX_V_constantXfalseX_aggregateXaX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", true, Operator.and).V().constant(false).aggregate("a").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | false |
+
+  Scenario: g_withSideEffectXa_true_andX_V_constantXfalseX_aggregateXlocal_aX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", true, Operator.and).V().constant(false).aggregate(Scope.local, "a").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | false |
+
+  Scenario: g_withSideEffectXa_true_orX_V_constantXfalseX_aggregateXaX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", true, Operator.or).V().constant(false).aggregate("a").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | true |
+
+  Scenario: g_withSideEffectXa_true_orX_V_constantXfalseX_aggregateXlocal_aX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", true, Operator.or).V().constant(false).aggregate(Scope.local, "a").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | true |
+
+  Scenario: g_withSideEffectXa_l1_addAllX_V_aggregateXaX_byXageX_capXaX
+    Given the modern graph
+    And using the parameter l1 defined as "l[d[1].i,d[2].i,d[3].i]"
```

Comment:
`l1` is actually used to indicate lambda parameters, this and similar tests below, should be changed to `And using the parameter xx1`.

Reply:
Changed all `l1` to `xx1`. Please check